### PR TITLE
[DOCS] Namespace redesign in documentation

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Introducing changes to code examples and content of articles with regard to implementation of `import openvino as ov`. This PR addresses CVS-110859.

### CODE REVIEW

File: docs/HOWTO/mri_reconstruction_demo.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import cv2 as cv
 import argparse
 import time
-from openvino.inference_engine import IECore
```
Comment (ilya-lavrenov) [inline docs/HOWTO/mri_reconstruction_demo.py:9]: this looks like a really old file, because we don't expose old API (inference_engine) in our documentation
maybe we need to completely remove it? if we don't have any references to it

File: docs/install_guides/troubleshooting-steps.md [hunk 2]
```diff
@@ -23,8 +23,8 @@ Check the versions of OpenVINO Runtime and Development Tools
           .. code-block:: py
              :force:
 
-             from openvino.tools.mo import convert_model
-             ov_model = convert_model(version=True)
+             import openvino as ov
+             ov_model = ov.convert_model(version=True)
```
Comment (ilya-lavrenov) [inline docs/install_guides/troubleshooting-steps.md:27]: will `ovc -h` print version?
or maybe even `ovc --version` ?

CC @popovaan

File: docs/optimization_guide/nncf/ptq/code/ptq_aa_openvino.py [hunk 3]
```diff
@@ -18,10 +18,10 @@ def transform_fn(data_item):
 #! [validation]
 import numpy as np
 import torch
-import openvino
+import openvino as ov
 from sklearn.metrics import accuracy_score
 
-def validate(model: openvino.runtime.CompiledModel, 
+def validate(model: ov.runtime.CompiledModel,
```
Comment (ilya-lavrenov) [inline docs/optimization_guide/nncf/ptq/code/ptq_aa_openvino.py:24]:
```suggestion
def validate(model: ov.CompiledModel, 
```

File: docs/snippets/cpu/compile_model.py [hunk 4]
```diff
@@ -3,15 +3,15 @@
...
-compiled_model = core.compile_model(model, "CPU")
+compiled_model = ov.compile_model(model, "CPU")
 #! [compile_model_default]
 
 #! [compile_model_multi]
-core = Core()
+core = ov.Core()
 model = core.read_model("model.xml")
-compiled_model = core.compile_model(model, "MULTI:CPU,GPU.0")
+compiled_model = ov.compile_model(model, "MULTI:CPU,GPU.0")
```
Comment (ilya-lavrenov) [inline docs/snippets/cpu/compile_model.py:16]:
```suggestion
compiled_model = core.compile_model(model, "MULTI:CPU,GPU.0")
```

File: docs/snippets/cpu/compile_model.py [hunk 5]
```diff
@@ -3,15 +3,15 @@
...
 
 #! [compile_model_default]
-from openvino.runtime import Core
+import openvino as ov
 
-core = Core()
+core = ov.Core()
 model = core.read_model("model.xml")
-compiled_model = core.compile_model(model, "CPU")
+compiled_model = ov.compile_model(model, "CPU")
```
Comment (ilya-lavrenov) [inline docs/snippets/cpu/compile_model.py:10]:
```suggestion
compiled_model = core.compile_model(model, "CPU")
```
Don't mix with `convert_model` :)

File: docs/snippets/cpu/multi_threading.py [hunk 6]
```diff
@@ -12,18 +11,18 @@
...
 
 # Use one logical processor per CPU core for inference when hyper threading is on
-compiled_model_3 = core.compile_model(model=model, device_name=device_name, config={properties.hint.enable_hyper_threading(False)})
+compiled_model_3 = ov.compile_model(model=model, device_name=device_name, config={ov.properties.hint.enable_hyper_threading(False)})
 # ! [ov:intel_cpu:multi_threading:part0]
 
 # ! [ov:intel_cpu:multi_threading:part1]
 # Disable CPU threads pinning for inference when system supoprt it
-compiled_model_4 = core.compile_model(model=model, device_name=device_name, config={properties.hint.enable_cpu_pinning(False)})
+compiled_model_4 = ov.compile_model(model=model, device_name=device_name, config={ov.properties.hint.enable_cpu_pinning(False)})
```
Comment (ilya-lavrenov) [inline docs/snippets/cpu/multi_threading.py:25]: please, revert to `core`

File: docs/snippets/cpu/ov_sparse_weights_decompression.py [hunk 7]
```diff
@@ -2,14 +2,14 @@
...
+import openvino as ov
 
 device_name = 'CPU'
 xml_path = 'model.xml'
 # ! [ov:intel_cpu:sparse_weights_decompression:part0]
 core = ov.Core()
 core.set_property("CPU", ov.properties.intel_cpu.sparse_weights_decompression_rate(0.8))
 model = core.read_model(model=xml_path)
-compiled_model = core.compile_model(model=model, device_name=device_name)
+compiled_model = ov.compile_model(model=model, device_name=device_name)
```
Comment (ilya-lavrenov) [inline docs/snippets/cpu/ov_sparse_weights_decompression.py:13]: revert

File: docs/snippets/ie_common.py [hunk 8]
```diff
@@ -5,8 +5,8 @@
 
 #! [ie:create_core]
 import numpy as np
-import openvino.inference_engine as ie
-core = ie.IECore()
+import openvino as ov
+core = ov.inference_engine.IECore()
```
Comment (ilya-lavrenov) [inline docs/snippets/ie_common.py:9]: let's keep old API code as is (it will be removed in 2024.0)

File: docs/snippets/cpu/compile_model.py (hunk 5 above)
Comment (sgolebiewski-intel) [reply docs/snippets/cpu/compile_model.py:10]: Hi, so I assume all other methods listed in init:
https://github.com/sgolebiewski-intel/openvino/blob/master/src/bindings/python/src/openvino/__init__.py

are also to be reverted?

@jiwaszki Can you relate?

File: docs/HOWTO/mri_reconstruction_demo.py (hunk 1 above)
Comment (sgolebiewski-intel) [reply docs/HOWTO/mri_reconstruction_demo.py:9]: We do not have any references to it apparently, so I will remove it.

File: docs/snippets/ie_common.py (hunk 8 above)
Comment (sgolebiewski-intel) [reply docs/snippets/ie_common.py]: Reverted it with the https://github.com/openvinotoolkit/openvino/pull/19084/commits/b4ead7997cfa46f091d33c5ca3cb3d3dbcd207a5 commit

File: docs/snippets/cpu/compile_model.py (hunk 5 above)
Comment (ilya-lavrenov) [reply docs/snippets/cpu/compile_model.py:10]: good question. I really forgot that we had `compile_model` without `Core` in Python API.
@slyalin @jiwaszki what is your preference here? and why?

File: docs/install_guides/troubleshooting-steps.md (hunk 2 above)
Comment (slyalin) [reply docs/install_guides/troubleshooting-steps.md:27]: ov.convert_model doesn't have `version` parameter. Use `ov.__version__`.

`ovc --version` gives output like this: `OpenVINO Model Converter (ovc) 2023.1.0-12028-dab01d286e6`

Comment (slyalin) [reply docs/install_guides/troubleshooting-steps.md:27]: Exactly when testing of snippets would help.

File: docs/snippets/cpu/compile_model.py (hunk 5 above)
Comment (slyalin) [reply docs/snippets/cpu/compile_model.py:10]: `ov.compile_model` is my preference. Top of the top. Main function in OpenVINO.

Comment (slyalin) [reply docs/snippets/cpu/compile_model.py:10]: The "only" problem with `ov.compile_model` is that it is still not aligned with `Core.compile_model`, you cannot pass device name -- don't know why, in my opinion it should be aligned.

The first example that should demonstrate `ov.compile_model` is `ov.compile_model('model.xml')` -- no need to use `read_model` in these particular snippets.

Comment (slyalin) [reply docs/snippets/cpu/compile_model.py:10]: > it is still not aligned with `Core.compile_model`, you cannot pass device name

That's of the reasons why most of the changes in this PR won't work.

Comment (sgolebiewski-intel) [reply docs/snippets/cpu/compile_model.py:10]: > > it is still not aligned with `Core.compile_model`, you cannot pass device name
> 
> That's one of the reasons why most of the changes in this PR won't work.

You mean changes regarding `compile_model` method, or in general? Ultimately, `ov.compile_model` should be reverted to `core.compile_model` to alleviate this particular issue?

File: docs/install_guides/troubleshooting-steps.md (hunk 2 above)
Comment (sgolebiewski-intel) [inline docs/install_guides/troubleshooting-steps.md:27]:
```suggestion
             ov_model = ov.convert_model(ov.__version__=True)
```

File: docs/snippets/cpu/compile_model.py (hunk 5 above)
Comment (ilya-lavrenov) [reply docs/snippets/cpu/compile_model.py:10]: @jiwaszki should we add one more overload of `compile_model` with device name? Current one assumes `AUTO` I suppose

Comment (jiwaszki) [reply docs/snippets/cpu/compile_model.py:10]: > @jiwaszki should we add one more overload of `compile_model` with device name? Current one assumes `AUTO` I suppose

This exact way was requested long time ago (ticket ref: 66277 and PR: https://github.com/openvinotoolkit/openvino/pull/8603).

However, It is possible to update the signature with additional `device` argument. It will be just `ov.compile_model(path, device="AUTO")`. It is change in API so it will be safer to do in the next release.

> @slyalin @jiwaszki what is your preference here? and why?

My preference is creation of `Core` manually as it has some other properties available. @ilya-lavrenov "I really forgot that we had compile_model without Core in Python API." -- is the best way to describe my preference.

File: docs/MO_DG/prepare_model/MO_Python_API.md [hunk 9]
```diff
@@ -73,9 +78,9 @@ Example of using ``InputCutInfo`` to freeze an input with value:
 .. code-block:: py
    :force:
 
-   from openvino.tools.mo import convert_model, InputCutInfo
-   
-   ov_model = convert_model(model, input=InputCutInfo("input_name", [3], np.float32, [0.5, 2.1, 3.4]))
+   import openvino as ov
+
+   ov_model = ov.convert_model(model, input=ov.InputCutInfo("input_name", [3], np.float32, [0.5, 2.1, 3.4]))
```
Comment (jiwaszki) [inline docs/MO_DG/prepare_model/MO_Python_API.md:83]:
```suggestion
   from openvino.tools.ovc import InputCutInfo

   ov_model = ov.convert_model(model, input=InputCutInfo("input_name", [3], np.float32, [0.5, 2.1, 3.4]))
```
I think you need to import it directly.

File: docs/MO_DG/prepare_model/MO_Python_API.md [hunk 10]
```diff
@@ -93,7 +98,10 @@ Example of using lists to set shapes, types and layout for multiple inputs:
 .. code-block:: py
    :force:
 
-   ov_model = convert_model(model, input=[("input1", [1,3,100,100], np.float32), ("input2", [1,3,100,100], np.float32)], layout=[Layout("NCHW"), LayoutMap("NCHW", "NHWC")])
+   import openvino as ov
+   from openvino.tools.mo import LayoutMap
+
+   ov_model = ov.convert_model(model, input=[("input1", [1,3,100,100], np.float32), ("input2", [1,3,100,100], np.float32)], layout=[ov.Layout("NCHW"), LayoutMap("NCHW", "NHWC")])
```
Comment (jiwaszki) [inline docs/MO_DG/prepare_model/MO_Python_API.md:104]: What about `from openvino.tools.mo import LayoutMap` and it's usage later. @slyalin `ov.convert_model` is ovc-based, what happens when mo-based arguments are passed around?

File: docs/MO_DG/prepare_model/MO_Python_API.md [hunk 11]
```diff
@@ -116,8 +123,9 @@ Example of using the ``LayoutMap`` class to change the layout of a model input:
 .. code-block:: py
    :force:
 
-   from openvino.tools.mo import convert_model, LayoutMap
-   
+   import openvino as ov
+   from openvino.tools.mo import LayoutMap
```
Comment (jiwaszki) [inline docs/MO_DG/prepare_model/MO_Python_API.md:127]: Same here.

File: docs/snippets/ov_extensions.py [hunk 12]
```diff
@@ -2,10 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-import openvino.runtime as ov
+import openvino as ov
 
 #! [py_frontend_extension_ThresholdedReLU_header]
-import openvino.runtime.opset8 as ops
```
Comment (jiwaszki) [inline docs/snippets/ov_extensions.py:8]: It needs to be done this way. Opsets are not exposed into top-level namespace.

File: docs/snippets/ov_extensions.py [hunk 13]
```diff
@@ -24,26 +23,23 @@
...
 #! [py_frontend_extension_ThresholdedReLU]
 def conversion(node):
     input_node = node.get_input(0)
     input_type = input_node.get_element_type()
-    greater = ops.greater(input_node, ops.constant([node.get_attribute("alpha")], input_type))
-    casted = ops.convert(greater, input_type.get_type_name())
-    return ops.multiply(input_node, casted).outputs()
+    greater = ov.opset8.greater(input_node, ov.opset8.constant([node.get_attribute("alpha")], input_type))
+    casted = ov.opset8.convert(greater, input_type.get_type_name())
+    return ov.opset8.multiply(input_node, casted).outputs()
```
Comment (jiwaszki) [inline docs/snippets/ov_extensions.py:35]: Won't work (please look at comment about `import openvino.runtime.opset8 as ops`)

File: docs/snippets/ov_infer_request.py [hunk 14]
```diff
@@ -3,12 +3,12 @@
...
 import numpy as np
 #! [import]
-import openvino.runtime as ov
+import openvino as ov
 #! [import]
 
 core = ov.Core()
 model = core.read_model("model.xml")
-compiled_model = core.compile_model(model, "AUTO")
+compiled_model = ov.compile_model(model, "AUTO")
```
Comment (jiwaszki) [inline docs/snippets/ov_infer_request.py:11]: For example, here you either go with `Core` path or `ov.compile_model` as the second one is actually reading model.

File: docs/snippets/ov_network_state_intro.py [hunk 15]
```diff
@@ -5,19 +5,18 @@
...
-    add = ops.add(read, input)
-    save = ops.assign(add, "variable0")
-    result = ops.result(add)
-    model = Model(results=[result], sinks=[save], parameters=[input])
+    input = ov.opset10.parameter([1, 1], dtype=np.float32)
+    read = ov.opset10.read_value(input, "variable0")
+    add = ov.opset10.add(read, input)
+    save = ov.opset10.assign(add, "variable0")
+    result = ov.opset10.result(add)
+    model = ov.Model(results=[result], sinks=[save], parameters=[input])
```
Comment (jiwaszki) [inline docs/snippets/ov_network_state_intro.py:19]: Same story as for opset8, needs to be imported on top or use `ov.runtime.opset10...`

File: docs/MO_DG/prepare_model/MO_Python_API.md (hunk 10 above)
Comment (slyalin) [reply docs/MO_DG/prepare_model/MO_Python_API.md:104]: `ovc` doesn't have `layout` parameter. This snippet is no more relevant to the top-level API we have. Please avoid modifying snippets in MO part of the docs we are going to keep it as is including `openvino.tools.mo.convert_model`.

File: docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md [hunk 16]
```diff
@@ -67,8 +67,8 @@ To get the full list of conversion parameters, run the following command:
        .. code-block:: py
           :force:
 
-          from openvino.tools.mo import convert_model
-          ov_model = convert_model(help=True)
+          import openvino as ov
+          ov_model = ov.convert_model(help=True)
```
Comment (popovaan) [inline docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md:71]: please revert this change.
Most of openvino.tools.**mo**.convert_model() examples won't work for openvino.convert_model().

File: docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md [hunk 17]
```diff
@@ -114,8 +114,8 @@ Below is a list of separate examples for different frameworks and model conversi
           .. code-block:: py
              :force:
 
-             from openvino.tools.mo import convert_model
-             ov_model = convert_model("BERT", input_shape=[[2,30],[2,30],[2,30]])
+             import openvino as ov
+             ov_model = ov.convert_model("BERT", input_shape=[[2,30],[2,30],[2,30]])
```
Comment (popovaan) [inline docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md:118]: Please revert changes in all convert_model() examples. 
This will not work.

Same comment for other convert_model() examples.

File: docs/MO_DG/prepare_model/MO_Python_API.md (hunk 9 above)
Comment (popovaan) [reply docs/MO_DG/prepare_model/MO_Python_API.md:83]: This will not work, we removed InputCutInfo from ovc.
Please import it from openvino.tools.mo

File: docs/MO_DG/prepare_model/MO_Python_API.md (hunk 11 above)
Comment (popovaan) [reply docs/MO_DG/prepare_model/MO_Python_API.md:127]: No need to change it. 
@sgolebiewski-intel please revert this change.

**Review by popovaan — CHANGES_REQUESTED** [review]

**Comment by popovaan** [comment]
Please do not change convert_model() examples. 
InputCutInfo, LayoutMap and convert_model() should be imported from openvino.tools.mo in all existing examples.

File: docs/MO_DG/prepare_model/MO_Python_API.md (hunk 10 above)
Comment (sgolebiewski-intel) [reply docs/MO_DG/prepare_model/MO_Python_API.md]: I have reverted them in the https://github.com/openvinotoolkit/openvino/pull/19084/commits/24d1321aa23402e051f7925957e6eedaacd7d9ff commit.

File: docs/MO_DG/prepare_model/MO_Python_API.md (hunk 11 above)
Comment (sgolebiewski-intel) [reply docs/MO_DG/prepare_model/MO_Python_API.md]: I have reverted it in the https://github.com/openvinotoolkit/openvino/pull/19084/commits/24d1321aa23402e051f7925957e6eedaacd7d9ff commit.

File: docs/snippets/ov_extensions.py (hunk 12 above)
Comment (sgolebiewski-intel) [reply docs/snippets/ov_extensions.py:8]: Have changed it in the https://github.com/openvinotoolkit/openvino/pull/19084/commits/5b7da5dc2ee73f1c39be7215b6771c39830be92b commit.

File: docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md (hunk 17 above)
Comment (sgolebiewski-intel) [reply docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md]: I have reverted them in the https://github.com/openvinotoolkit/openvino/pull/19084/commits/24d1321aa23402e051f7925957e6eedaacd7d9ff commit.

**Comment by sgolebiewski-intel** [comment]
Closing in favor of an updated one: https://github.com/openvinotoolkit/openvino/pull/19365
